src/unix/bsd/apple/mod.rs

```diff
@@ -23,6 +23,10 @@ pub type nl_item = ::c_int;
 pub type id_t = ::c_uint;
 pub type sem_t = ::c_int;
 pub type idtype_t = ::c_uint;
+pub type kevent_nevents = ::c_int;
+pub type kevent_filter = ::int16_t;
+pub type kevent_flags = ::uint16_t;
+pub type kevent_fflags = ::uint32_t;
```

Comment:
Hm if these don't actually exist, how come we're defining them?

Reply:
Well, the idea is that, for example, `kevent_nevents` is "the type that the `kevent` function uses for the `nevents` param on this BSD variant" (it varies). So that in user code I can cast a variable `as kevent_nevents` without having to worry about which particular BSD I'm currently on. Same for the other `kevent_*` types.

Reply:
That makes sense, yes, but it's not what libc does. We don't add things that aren't in headers.